docs/requirements/database-structure-for-plan-entities.md

```diff
@@ -0,0 +1,304 @@
+- Each day (`plan_day`) has **1–5 tasks**, but the data model should allow any number of tasks per day.
+- Tasks contain:
+  - Plain text **title** and **description**.
+  - Optional **tip**.
+  - **Execution time** (morning, afternoon, evening).
+  - **Order** within the day.
+  - **Timestamps** for updates and completion.
+- Tasks can be marked as **complete/incomplete**.
+- Tasks do not have deadlines or unlock timers.
+
+---
+
+## 4. Progress & History
+The system tracks:
+- **History of checklists** should be saved.
+- **History of plans**, so a user can return to and update previous plans.
+
+# Conceptual Data Model
+
+```mermaid
+erDiagram
+  users ||--o{ plan : "has"
+  users {
+      int id PK
+      dateTime created_at
+      dateTime updated_at
+      varchar email
+      varchar name
+      text password_hash
+      text password_salt
+  }
+
+  intensity ||--o{ duration : "has"
+  intensity o|--|| level_type : "enum:level_type"
+  intensity {
+      int id PK
+      enum level_type
+      int min_tasks_per_day
+      int max_tasks_per_day
+  }
+
+  level_type {
+      low low
+      medium medium
+      high high
+  }
+
+  duration ||--o{ plan : "defines"
+  duration {
+      int id PK
+      int number_of_day
+      int intensity_id FK
+  }
+
+  plan ||--|{ plan_day : "has"
+  plan ||--o| plan : "regenerated from"
+  plan {
+      int id PK
+      varchar title
+      int user_id FK
+      int duration_id FK
+      int regenerated_from_id FK
+      boolean is_active
+      dateTime created_at
+  }
+
+  plan_day ||--|{ task : "has"
+  plan_day {
+      int id PK
+      int day_number
+      boolean is_regenerated
+      int plan_id FK
+  }
+
+  task {
+      int id PK
+      varchar title
+      text description
+      int order
+      int plan_day_id FK
+      boolean is_completed
+      int execution_time_id FK
+      dateTime updated_at
+      dateTime completed_at
+  }
+
+  execution_time ||--o{ task : "used in"
+  execution_time o|--|| execution_time_type : "enum:execution_time_type"
+  execution_time {
+      int id PK
+      enum execution_time_type
+      time start_time
+      time end_time
+  }
+
+  execution_time_type {
+      morning morning
+      afternoon afternoon
+      evening evening
+  }
+```
+
+# Physical Data Model
+
+```
+CREATE TYPE level_type AS ENUM ('low', 'medium', 'high');
+CREATE TYPE execution_time_type AS ENUM ('morning', 'afternoon', 'evening');
+
+CREATE TABLE intensity (
+  id SERIAL PRIMARY KEY,
+  level_type level NOT NULL,
+  min_tasks_per_day INT CHECK (min_tasks_per_day > 0) NOT NULL,
+  max_tasks_per_day INT CHECK (max_tasks_per_day >= min_tasks_per_day) NOT NULL,
+  created_at TIMESTAMPTZ DEFAULT CURRENT_TIMESTAMP NOT NULL
+);
+
+CREATE TABLE duration (
+  id SERIAL PRIMARY KEY,
+  number_of_days INT CHECK (number_of_days > 0) NOT NULL,
+  intensity_id INT NOT NULL REFERENCES intensity(id) ON DELETE RESTRICT,
+  created_at TIMESTAMPTZ DEFAULT CURRENT_TIMESTAMP NOT NULL
+);
```

Comment:
After getting all approvals and porting the DDL script to an ORM, please remove this DDL script from here as it will be documented inside the ORM script